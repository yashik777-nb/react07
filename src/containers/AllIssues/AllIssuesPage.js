import React from "react";
import IssuesAPI from "../../data/IssuesAPI";
import IssuesList from "./IssuesList";

export default class AllIssuesPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      issues: [],
    };
  }

  render() {
    return (
      <div>
        <h1 style={{ marginLeft: "10px" }}>IssuesList</h1>
        <IssuesList issues={this.state.issues} />;
      </div>
    );
  }

  componentDidMount() {
    this.setState({ issues: IssuesAPI.getAllIssues() });
  }
}
