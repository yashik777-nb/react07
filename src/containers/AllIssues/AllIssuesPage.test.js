import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import AllIssuesPage from "./AllIssuesPage";

configure({ adapter: new Adapter() });

describe("All Issues Page Test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AllIssuesPage />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("All Issue Page should have h1", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });

  it("All Issue Page should have Issues List", () => {
    expect(wrapper.find("IssuesList").length).toEqual(1);
  });

  test("All Issues Page Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
