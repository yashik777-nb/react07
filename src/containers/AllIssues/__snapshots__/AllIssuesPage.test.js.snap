// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Issues Page Test All Issues Page Snapshot 1`] = `ReactWrapper {}`;
